### PR TITLE
patch preprocess from contributor api

### PR DESCRIPTION
**Pull request by azime**

### CODE REVIEW

File: documentation/openAPI.yaml [hunk 1]
```diff
@@ -757,6 +757,34 @@ paths:
...
+        - application/json
+      parameters:
+        - in: path
+          name: contributor_id
+          type: string
+          description: Identifier of the contributor
+          required: true
+      responses:
+        200:
+          description: Requested contributor data source
```
Comment (nberard) [inline documentation/openAPI.yaml]: Requested contributor jobs

File: documentation/openAPI.yaml [hunk 2]
```diff
@@ -757,6 +757,34 @@ paths:
...
+          schema:
+            $ref: '#/definitions/job_list'
+        400:
```
Comment (nberard) [inline documentation/openAPI.yaml]: there seems to be no 400 raised

File: tartare/interfaces/contributors.py [hunk 3]
```diff
@@ -30,32 +30,56 @@
...
+            "data_sources": schema.DataSourceSchema,
+            "preprocesses": schema.PreProcessSchema
+        }
+        existing_id = [d.id for d in source]
+        logging.getLogger(__name__).debug("PATCH : list of existing {} ids {}".format(key, str(existing_id)))
+        # constructing PATCH data
+        patched_data = None
+        if key in request_data:
+            patched_data = map_model.get(key)(many=True).dump(source).data
+            for ds in request_data[key]:
```
Comment (nberard) [inline tartare/interfaces/contributors.py]: ds as variable name is confusing (data_sources, preprocesses)

File: tartare/interfaces/contributors.py [hunk 4]
```diff
@@ -30,32 +30,56 @@
...
+                if ds['id'] in existing_id:
+                    pds = next((p for p in patched_data if p['id'] == ds['id']), None)
```
Comment (nberard) [inline tartare/interfaces/contributors.py]: same for pds very hard to follow ^^

File: documentation/openAPI.yaml (hunk 2 above)
Comment (azime) [inline documentation/openAPI.yaml]: :+1:
